src/key_stores/merge_key_store.ts

```diff
@@ -35,13 +35,16 @@ import { KeyPair } from '../utils/key_pair';
  * ```
  */
 export class MergeKeyStore extends KeyStore {
+    private writeIndex: number;
     keyStores: KeyStore[];
 
     /**
-     * @param keyStores first keystore gets all write calls, read calls are attempted from start to end of array
+     * @param keyStores read calls are attempted from start to end of array
+     * @param writeIndex the keystore index that will receive all write calls
      */
-    constructor(keyStores: KeyStore[]) {
+    constructor(keyStores: KeyStore[], writeIndex = 0) {
```

Comment:
I think we should pass `writeKeyStore` instance explicitly instead of passing index

Reply:
as if there is a number passed as an argument I have to go an look to docs, etc to understand what this number represents vs if you pass keystore directly it seems to be more intuitive 

though maybe it's just me

Reply:
yes good call.

Reply:
Ok, I've experimented with this a bit. It may be a bit awkward because it makes it so the `writeKeyStore` may not be in the list of `keyStores`:

We can check for this:

```ts
config.keyStore = new MergeKeyStore([keyPathStore, (config.keyStore || config.deps.keyStore)], keyPathStore);
```

```ts
    constructor(keyStores: KeyStore[], writeKeyStore?: KeyStore) {
        super();
        // probably best to check that writeKeyStore is in keystores
        if(!keyStores.some(ks => ks === writeKeyStore)) {
            throw new Error('keyStores is missing writeKeyStore');
        }

        this.writeKeyStore = writeKeyStore || keyStores[0];
        this.keyStores = keyStores;
    }
```

Maybe the best solution is to make the second param to the constructor an options object so we can:
 
```ts
config.keyStore = new MergeKeyStore([keyPathStore, (config.keyStore || config.deps.keyStore)], { writeKeyStoreIndex: 1 });
```